### PR TITLE
Convert settings page [ng2]

### PR DESCRIPTION
**Pull request by ruddell**
First try at angular-2 so there will probably be some corrections

### CODE REVIEW

File: generators/client-2/templates/src/main/webapp/app/account/settings/settings.html [hunk 1]
```diff
@@ -1,83 +1,86 @@
...
+                </div>
+                <%_ if (enableTranslation){ _%>
+                <div class="form-group" *ngIf="languages && languages.length > 0">
                     <label for="langKey" translate="settings.form.language">Language</label>
-                    <select id="langKey" name="langKey" class="form-control" ng-model="vm.settingsAccount.langKey" ng-controller="<%=jhiPrefixCapitalized%>LanguageController as languageVm" ng-options="code as (code | findLanguageFromKey) for code in languageVm.languages"></select>
-                </div><% } %>
-
-                <button type="submit" ng-disabled="form.$invalid" class="btn btn-primary" translate="settings.form.button">Save</button>
+                    <select class="form-control" id="langKey" name="langKey" [(ngModel)]="settingsAccount.langKey">
+                        <option *ngFor="let language of languages" [value]="language">{{language}}</option>
```
Comment (ruddell) [inline generators/client-2/templates/src/main/webapp/app/account/settings/settings.html]: I couldn't figure out how to get the findLanguageFromKey pipe to work.  Any help is appreciated

File: generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import {Component, Inject, OnInit} from '@angular/core';
+import { <%=jhiPrefixCapitalized%>LanguageService } from '../../components/language/language.service';
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts]: this should be imported only when `enableTranslation` is true

File: generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    templateUrl: 'app/account/settings/settings.html'
+})
+export class SettingsComponent implements OnInit {
+    error: string;
+    success: string;
+    settingsAccount: any;
+    languages: any[];
+    Auth: any;
+    Principal: any;
+    <%=jhiPrefixCapitalized%>LanguageService: any;
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts]: use only when `enableTranslation` is true

File: generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    constructor(@Inject('$translate') private $translate, @Inject('Auth') Auth, @Inject('Principal') Principal,
+                private languageService: <%=jhiPrefixCapitalized%>LanguageService) {
+        this.Auth = Auth;
+        this.Principal = Principal;
+        this.<%=jhiPrefixCapitalized%>LanguageService = <%=jhiPrefixCapitalized%>LanguageService;
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/settings/_settings.component.ts]: same here

File: generators/client-2/templates/src/main/webapp/app/account/settings/settings.html [hunk 5]
```diff
@@ -1,83 +1,86 @@
...
 
-            <div class="alert alert-success" ng-show="vm.success" translate="settings.messages.success">
+            <div class="alert alert-success" *ngIf="success" translate="settings.messages.success">
                 <strong>Settings saved!</strong>
             </div>
 
-           <jhi-alert-error></jhi-alert-error>
+            <jhi-alert-error></jhi-alert-error>
 
-            <form name="form" role="form" novalidate ng-submit="vm.save()" show-validation>
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/settings/settings.html]: why did you remove `novalidate` and `show-validation`

File: generators/client-2/templates/src/main/webapp/app/account/settings/settings.html (hunk 1 above)
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/account/settings/settings.html]: The `findLanguageFromKey `pipe needs to be moved from the `app.ng2module` to `shared.ng2module` and exported from there so that it can be used system wide

**Comment by ruddell** [comment]
@deepu105 Thanks for the feedback.  I made the requested changes and the pipe is working now.  I also had to change the navbar component to get the TS to compile without translation.

I removed the novalidate and show-validation because my changes were modeled off of #4122 ([link to line](https://github.com/jhipster/generator-jhipster/pull/4122/files#diff-5a235814c91074a25dcf5fa7c8abf86bL25)).  I put them back into the form tag.
